pkg/server/k8s/client.go

```diff
@@ -625,6 +625,23 @@ func (k *Client) createIngress(namespace, appName string, vHosts []string) error
 	return errors.Wrap(err, "create ingress failed")
 }
 
+func (k *Client) UpdateIngress(namespace, name string, vHosts []string) error {
+	kc, err := k.buildClient()
+	if err != nil {
+		return err
+	}
+	old, err := kc.ExtensionsV1beta1().
+		Ingresses(namespace).
+		Get(name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "get ingress failed")
+	}
+	new := ingressSpec(namespace, name, vHosts)
+	old.Spec.Rules = new.Spec.Rules
+	_, err = kc.ExtensionsV1beta1().Ingresses(namespace).Update(old)
```

Comment:
Is it doesn't work if you only pass the new spec to `Update` method?

Reply:
It works but overwrites the tls section that might be configured manually.

Reply:
@drgarcia1986

Reply:
gotcha 👍